core/src/Streamly/Internal/Data/Stream/StreamD/Transform.hs

```diff
@@ -1895,7 +1895,8 @@ catEithers = fmap (either id id)
 -- separator elements determined by the supplied predicate, separator is
 -- considered as infixed between two segments:
 --
--- >>> splitOn' p xs = Stream.fold Fold.toList $ Stream.splitOn p Fold.toList (Stream.fromList xs)
+-- >>> import Streamly.Internal.Data.Stream.StreamD.Transform (splitOn)
+-- >>> splitOn' p xs = Stream.fold Fold.toList $ splitOn p Fold.toList (Stream.fromList xs)
```

Comment:
Should use Stream.splitOn . The example should work if someone uses the imports as shown in the haddock docs in GHCi.